pkg/approver/manager/subjectaccessreview.go

```diff
@@ -24,54 +24,43 @@ import (
+	"github.com/cert-manager/policy-approver/pkg/approver"
 )
 
-// Manager is responsible for evaluating whether incoming CertificateRequests
-// should be approved or denied, checking CertificateRequestPolicys against
-// evaluators which have been registries. Policies will be chosen based on
-// their suitability for a particular request, namely whether they are bound to
-// a policy via RBAC.
-type Manager struct {
-	client.Client
-	approveWhenNoPolicies bool
+// subjectaccessreview is a manager that calls evaluators with
+// CertificateRequestPolicys that have been RBAC bound to the user who appears
+// in the passed CertificateRequest to Evaluate.
+type subjectaccessreview struct {
+	client client.Client
 
-	evaluators registry.Registry
+	evaluators []approver.Evaluator
 }
 
-// NewManager constructs a new policy Manager evaluates whether
-// CertificateRequests should be approved or denied, managing registered
-// evaluators.
-func NewManager(client client.Client, approveWhenNoPolicies bool) *Manager {
-	return &Manager{
-		Client:                client,
-		approveWhenNoPolicies: approveWhenNoPolicies,
-		evaluators:            registry.List(),
+// NewSubjectAccessReview constructs a new approver Manager which evaluates
+// whether CertificateRequests should be approved or denied, managing
+// registered evaluators.
+func NewSubjectAccessReview(client client.Client, evaluators []approver.Evaluator) *subjectaccessreview {
+	return &subjectaccessreview{
+		client:     client,
+		evaluators: evaluators,
 	}
 }
 
-// Evaluate will evaluate whether the incoming CertificateRequest should be
-// approved.
-// - Consumers should consider a true response meaning the CertificateRequest
-//   is **approved**.
-// - Consumers should consider a false response and no error to mean the
-//   CertificateRequest is **denied**.
-// - Consumers should treat any error response as marking the
-//   CertificateRequest as neither approved nor denied, and may consider
-//   re-evaluation at a later time.
-func (m *Manager) Evaluate(ctx context.Context, cr *cmapi.CertificateRequest) (bool, PolicyMessage, error) {
+// Review will evaluate whether the incoming CertificateRequest should be
+// approved. All evaluators will be called with CertificateRequestPolicys that
+// have been RBAC bound to the user included in the CertificateRequest.
+func (s *subjectaccessreview) Review(ctx context.Context, cr *cmapi.CertificateRequest) (bool, string, error) {
 	crps := new(cmpapi.CertificateRequestPolicyList)
-	if err := m.List(ctx, crps); err != nil {
+	if err := s.client.List(ctx, crps); err != nil {
 		return false, "", err
 	}
 
 	// If no CertificateRequestPolicys exist, exit early approved if configured
 	// to do so
-	if m.approveWhenNoPolicies && len(crps.Items) == 0 {
-		return true, MessageNoExistingCertificateRequestPolicy, nil
-	}
+	//if m.approveWhenNoPolicies && len(crps.Items) == 0 {
+	//	return true, MessageNoExistingCertificateRequestPolicy, nil
+	//}
```

Comment:
is this intentionally commented out?

Reply:
It is, we will no longer be denying requests if no policies exist.

Reply:
Personally I'd say that this would ideally have an explanatory comment if the commented code is to remain.

Reply:
Fair enough, I'll delete it.